filters/include/pcl/filters/impl/passthrough.hpp

```diff
@@ -1,140 +1,59 @@
+    const auto* pt_data = reinterpret_cast<const std::uint8_t*>(&cloud[idx]);
+    float field_value = 0;
+    memcpy(&field_value, pt_data + fields_[distance_idx_].offset, sizeof(float));
+    if (std::isfinite(field_value) && field_value >= filter_limit_min_ &&
+        field_value <= filter_limit_max_) {
+      return true;
     }
   }
-  else
-  {
-    // Attempt to get the field name's index
-    std::vector<pcl::PCLPointField> fields;
-    int distance_idx = pcl::getFieldIndex<PointT> (filter_field_name_, fields);
-    if (distance_idx == -1)
-    {
-      PCL_WARN ("[pcl::%s::applyFilter] Unable to find field name in point type.\n", getClassName ().c_str ());
-      indices.clear ();
-      removed_indices_->clear ();
-      return;
-    }
-
-    // Filter for non-finite entries and the specified field limits
-    for (const auto ii : *indices_)  // ii = input index
-    {
-      // Non-finite entries are always passed to removed indices
-      if (!std::isfinite ((*input_)[ii].x) ||
-          !std::isfinite ((*input_)[ii].y) ||
-          !std::isfinite ((*input_)[ii].z))
-      {
-        if (extract_removed_indices_)
-          (*removed_indices_)[rii++] = ii;
-        continue;
-      }
-
-      // Get the field's value
-      const std::uint8_t* pt_data = reinterpret_cast<const std::uint8_t*> (&(*input_)[ii]);
-      float field_value = 0;
-      memcpy (&field_value, pt_data + fields[distance_idx].offset, sizeof (float));
-
-      // Remove NAN/INF/-INF values. We expect passthrough to output clean valid data.
-      if (!std::isfinite (field_value))
-      {
-        if (extract_removed_indices_)
-          (*removed_indices_)[rii++] = ii;
-        continue;
-      }
-
-      // Outside of the field limits are passed to removed indices
-      if (!negative_ && (field_value < filter_limit_min_ || field_value > filter_limit_max_))
-      {
-        if (extract_removed_indices_)
-          (*removed_indices_)[rii++] = ii;
-        continue;
-      }
-
-      // Inside of the field limits are passed to removed indices if negative was set
-      if (negative_ && field_value >= filter_limit_min_ && field_value <= filter_limit_max_)
-      {
-        if (extract_removed_indices_)
-          (*removed_indices_)[rii++] = ii;
-        continue;
-      }
+  return false;
+}
 
-      // Otherwise it was a normal point for output (inlier)
-      indices[oii++] = ii;
-    }
+template <typename PointT> void
+pcl::PassThrough<PointT>::applyFilter (Indices &indices)
+{
+  // Attempt to get the field name's index
+  if (!filter_field_name_.empty() && distance_idx_ == -1)
+  {
+  PCL_WARN ("[pcl::%s::applyFilter] Unable to find field name in point type.\n", getClassName ().c_str ());
+  indices.clear ();
+  removed_indices_->clear ();
+  return;
   }
 
-  // Resize the output arrays
-  indices.resize (oii);
-  removed_indices_->resize (rii);
+  const auto lambda = [this](const PointCloud& cloud, index_t idx) {
+    return condition(cloud, idx);
```

Comment:
Is the finite-check part of the condition or part of a separate filter. As in, which represents this better:
```cpp
return condition(cloud, idx);
// OR
return isXYZFinite(cloud[idx]) && condition(cloud, idx);
```
Because technically, `isXYZFinite(cloud[idx])` is a passthrough filter with min and max values `-FLT_MAX` and `FLT_MAX`. :thinking: